### PR TITLE
Bug/gh761 switch domain shows selected domain

### PR DESCRIPTION
**Pull request by omaratrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-IME-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-IME-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

Switch domain dialog shows currently selected domain instead of the default domain.

### CODE REVIEW

File: CDP4ShellDialogs/ViewModels/ModelIterationDomainSwitchDialogViewModel.cs [hunk 1]
```diff
@@ -108,6 +108,7 @@ private async Task ExecuteDomainSwitch()
                 if (openIteration != null)
                 {
                     session.SwitchDomain(modelrow.IterationIid, modelrow.SelectedDomain);
+                    modelrow.ActiveParticipant.SelectedDomain = modelrow.SelectedDomain;
```
Comment (alexatrhea) [inline CDP4ShellDialogs/ViewModels/ModelIterationDomainSwitchDialogViewModel.cs:111]: Why do you need this? The switching should already have happened and the dialog is presumably closing.

File: CDP4ShellDialogs/ViewModels/ModelSelectionIterationSetupRowViewModel.cs [hunk 2]
```diff
@@ -47,10 +47,12 @@ public ModelSelectionIterationSetupRowViewModel(IterationSetup iterationSetup, P
 
             if (participant.Domain.Count != 0)
             {
-                this.DomainOfExpertises.AddRange(participant.Domain.OrderBy(x => x.ShortName)); 
+                this.DomainOfExpertises.AddRange(participant.Domain.OrderBy(x => x.ShortName));
 
-                this.SelectedDomain = this.DomainOfExpertises.Contains(this.ActiveParticipant.Person.DefaultDomain)
-                    ? this.ActiveParticipant.Person.DefaultDomain
+                var selectedDomain2 = this.ActiveParticipant.SelectedDomain.Name;
```
Comment (alexatrhea) [inline CDP4ShellDialogs/ViewModels/ModelSelectionIterationSetupRowViewModel.cs:52]: Why do you need this? Not being used anywhere...

Comment (samatrhea) [inline CDP4ShellDialogs/ViewModels/ModelSelectionIterationSetupRowViewModel.cs:52]: I have the impression that `selectedDomain2` is set but never used, if that is indeed the case, please remove

File: CDP4ShellDialogs/ViewModels/ModelSelectionIterationSetupRowViewModel.cs [hunk 3]
```diff
@@ -47,10 +47,12 @@ public ModelSelectionIterationSetupRowViewModel(IterationSetup iterationSetup, P
...
+
+                this.SelectedDomain = this.DomainOfExpertises.Contains(this.ActiveParticipant.SelectedDomain)
```
Comment (samatrhea) [inline CDP4ShellDialogs/ViewModels/ModelSelectionIterationSetupRowViewModel.cs:54]: not sure why this code is changed, this is not related to the ticket?

**Review by samatrhea — CHANGES_REQUESTED** [review]

File: CDP4ShellDialogs/ViewModels/ModelIterationDomainSwitchDialogViewModel.cs [hunk 4]
```diff
@@ -49,17 +49,17 @@ public ModelIterationDomainSwitchDialogViewModel(IEnumerable<ISession> sessionAv
...
+        public DisposableReactiveList<SwitchDomainSessionRowViewModel> SessionsAvailable { get; private set; }
 
         /// <summary>
         /// Gets the list of <see cref="IterationSetup"/> selected
         /// </summary>
-        public ReactiveList<IViewModelBase<Thing>> SelectedIterations { get; private set; }
+        public ReactiveList<IViewModelBase<Thing>> SelectedIterations { get; set; }
 
         /// <summary>
         /// Gets the Select <see cref="ICommand"/>
```
Comment (alexatrhea) [inline CDP4ShellDialogs/ViewModels/ModelIterationDomainSwitchDialogViewModel.cs:60]: Correct the documentation pls.

File: CDP4ShellDialogs/ViewModels/ModelIterationDomainSwitchDialogViewModel.cs [hunk 5]
```diff
@@ -49,17 +49,17 @@ public ModelIterationDomainSwitchDialogViewModel(IEnumerable<ISession> sessionAv
         /// <summary>
         /// Gets the list of <see cref="BaseRowViewModel"/> available
```
Comment (alexatrhea) [inline CDP4ShellDialogs/ViewModels/ModelIterationDomainSwitchDialogViewModel.cs:50]: Correct documentation.

File: CDP4ShellDialogs/ViewModels/SwitchDomainEngineeringModelSetupRowViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,68 @@
+﻿// -------------------------------------------------------------------------------------------------
+// <copyright file="ModelSelectionEngineeringModelSetupRowViewModel.cs" company="RHEA System S.A.">
```
Comment (alexatrhea) [inline CDP4ShellDialogs/ViewModels/SwitchDomainEngineeringModelSetupRowViewModel.cs:2]: incorrect name

Comment (alexatrhea) [reply CDP4ShellDialogs/ViewModels/SwitchDomainEngineeringModelSetupRowViewModel.cs:2]: Please regenerate the headers on all of the new files.

File: CDP4ShellDialogs/ViewModels/SwitchDomainEngineeringModelSetupRowViewModel.cs [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        /// </param>
+        public SwitchDomainEngineeringModelSetupRowViewModel(EngineeringModelSetup engineeringModelSetup, ISession session)
+            : base(engineeringModelSetup, session, null)
+        {
+            this.IterationSetupRowViewModels = new DisposableReactiveList<SwitchDomainIterationSetupRowViewModel>();
+            this.InitializeContainers();
+        }
+
+        /// <summary>
+        /// Gets the <see cref="IterationSetupRowViewModel"/> that are contained by the row-view-model
```
Comment (alexatrhea) [inline CDP4ShellDialogs/ViewModels/SwitchDomainEngineeringModelSetupRowViewModel.cs:37]: Use correct class name in doc.

File: CDP4ShellDialogs/ViewModels/SwitchDomainEngineeringModelSetupRowViewModel.cs [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+        /// </summary>
+        public DisposableReactiveList<SwitchDomainIterationSetupRowViewModel> IterationSetupRowViewModels { get; private set; }
+
+        /// <summary>
+        /// Initializes containers
+        /// </summary>
+        private void InitializeContainers()
```
Comment (alexatrhea) [inline CDP4ShellDialogs/ViewModels/SwitchDomainEngineeringModelSetupRowViewModel.cs:44]: Rename method to `InitializeContainedRows` and update the documentation.

File: CDP4ShellDialogs/ViewModels/SwitchDomainEngineeringModelSetupRowViewModel.cs [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+            var iterationSetups = this.Session.OpenIterations.Keys.Select(i => i.IterationSetup).Where(i => i.Container.Iid.Equals(this.Thing.Iid));
+
+            foreach(var iterationSetup in iterationSetups)
+            {
+                this.AddIteration(iterationSetup);
+            }
+        }
+
+        /// <summary>
+        /// Add the <see cref="IterationSetup"/> to the contained <see cref="IterationSetupRowViewModels"/>
```
Comment (alexatrhea) [inline CDP4ShellDialogs/ViewModels/SwitchDomainEngineeringModelSetupRowViewModel.cs:55]: Use correct class name for the row.

File: CDP4ShellDialogs/ViewModels/SwitchDomainEngineeringModelSetupRowViewModel.cs [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+        /// </summary>
+        /// <param name="iteration">
+        /// the <see cref="IterationSetup"/> object that are to be added
+        /// </param>
+        private void AddIteration(IterationSetup iteration)
```
Comment (alexatrhea) [inline CDP4ShellDialogs/ViewModels/SwitchDomainEngineeringModelSetupRowViewModel.cs:60]: variable should be `iterationSetup` and update name in doc.

File: CDP4ShellDialogs/ViewModels/SwitchDomainIterationSetupRowViewModel.cs [hunk 11]
```diff
@@ -0,0 +1,97 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
```
Comment (alexatrhea) [inline CDP4ShellDialogs/ViewModels/SwitchDomainIterationSetupRowViewModel.cs:1]: new header needed.

File: CDP4ShellDialogs/ViewModels/SwitchDomainIterationSetupRowViewModel.cs [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+    /// </summary>
+    public class SwitchDomainIterationSetupRowViewModel : IterationSetupRowViewModel
+    {
+        /// <summary>
+        /// backing field for <see cref="SelectedDomain"/>
+        /// </summary>
+        private DomainOfExpertise selectedDomain;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ModelIterationDomainSwitchDialogViewModel"/> class.
```
Comment (alexatrhea) [inline CDP4ShellDialogs/ViewModels/SwitchDomainIterationSetupRowViewModel.cs:27]: Update class name

File: CDP4ShellDialogs/ViewModels/SwitchDomainSessionRowViewModel.cs [hunk 13]
```diff
@@ -0,0 +1,75 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
```
Comment (alexatrhea) [inline CDP4ShellDialogs/ViewModels/SwitchDomainSessionRowViewModel.cs:1]: New header

File: CDP4ShellDialogs/ViewModels/SwitchDomainSessionRowViewModel.cs [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+    using CDP4Common.EngineeringModelData;
+    using CDP4Common.SiteDirectoryData;
+    using CDP4CommonView;
+
+    using CDP4Composition.Mvvm.Types;
+
+    using CDP4Dal;
+
+    /// <summary>
+    /// The purpose of the <see cref="SessionRowViewModel"/> is to represent a <see cref="Session"/> and its <see cref="SiteDirectory"/>
```
Comment (alexatrhea) [inline CDP4ShellDialogs/ViewModels/SwitchDomainSessionRowViewModel.cs:19]: Use correct class name.

File: CDP4ShellDialogs/ViewModels/SwitchDomainSessionRowViewModel.cs [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+    /// </summary>
+    public class SwitchDomainSessionRowViewModel : SiteDirectoryRowViewModel
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ModelSelectionSessionRowViewModel"/> class.
```
Comment (alexatrhea) [inline CDP4ShellDialogs/ViewModels/SwitchDomainSessionRowViewModel.cs:24]: Correct class name

File: CDP4ShellDialogs/ViewModels/SwitchDomainSessionRowViewModel.cs [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+                {
+                    this.AddModelRowViewModel(model);
+                }
+            }
+
+            this.UpdateProperties();
+        }
+
+        /// <summary>
+        /// Gets the <see cref="EngineeringModelSetupRowViewModel"/> that are contained by the row-view-model
```
Comment (alexatrhea) [inline CDP4ShellDialogs/ViewModels/SwitchDomainSessionRowViewModel.cs:53]: Use correct class name

File: CDP4ShellDialogs/ViewModels/SwitchDomainSessionRowViewModel.cs [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+        /// </summary>
+        public DisposableReactiveList<SwitchDomainEngineeringModelSetupRowViewModel> EngineeringModelSetupRowViewModels { get; private set; }
+        
+        /// <summary>
+        /// Add a <see cref="EngineeringModelSetupRowViewModel"/> to the list of <see cref="EngineeringModelSetup"/>s
+        /// </summary>
+        /// <param name="model">The <see cref="EngineeringModelSetup"/> that is to be added</param>
+        private void AddModelRowViewModel(EngineeringModelSetup model)
```
Comment (alexatrhea) [inline CDP4ShellDialogs/ViewModels/SwitchDomainSessionRowViewModel.cs:61]: parameter should be `engineeringModelSetup` and update the docs.

File: CDP4ShellDialogsTestFixture/ViewModels/ModelIterationDomainSwitchDialogViewModelTestFixture.cs [hunk 18]
```diff
@@ -0,0 +1,124 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
```
Comment (alexatrhea) [inline CDP4ShellDialogsTestFixture/ViewModels/ModelIterationDomainSwitchDialogViewModelTestFixture.cs:1]: Headers should be correct.

File: CDP4ShellDialogsTestFixture/ViewModels/SwitchDomainSessionTextFixture.cs [hunk 19]
```diff
@@ -0,0 +1,135 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="ModelSelectionSessionTestFixture.cs" company="RHEA System S.A.">
```
Comment (alexatrhea) [inline CDP4ShellDialogsTestFixture/ViewModels/SwitchDomainSessionTextFixture.cs:2]: Header

**Review by alexatrhea — CHANGES_REQUESTED** [review]
Please pay attention to correct naming and documentation.

File: CDP4ShellDialogs/ViewModels/ModelIterationDomainSwitchDialogViewModel.cs (hunk 4 above)
Comment (alexatrhea) [reply CDP4ShellDialogs/ViewModels/ModelIterationDomainSwitchDialogViewModel.cs:60]: This is not resolved. should be "Gets the Switch <see cref=" not Select

File: CDP4ShellDialogs/ViewModels/SwitchDomainEngineeringModelSetupRowViewModel.cs (hunk 9 above)
Comment (alexatrhea) [reply CDP4ShellDialogs/ViewModels/SwitchDomainEngineeringModelSetupRowViewModel.cs:55]: this is not resolved

File: CDP4ShellDialogs/ViewModels/SwitchDomainEngineeringModelSetupRowViewModel.cs (hunk 10 above)
Comment (alexatrhea) [reply CDP4ShellDialogs/ViewModels/SwitchDomainEngineeringModelSetupRowViewModel.cs:60]: not resolved

**Review by alexatrhea — CHANGES_REQUESTED** [review]

**Review by alexatrhea — APPROVED** [review]

**Pull request merged**